src/components/molecules/Visualizer/hooks.ts

```diff
@@ -243,20 +242,20 @@ function useLayers({
   selected: outerSelectedPrimitiveId,
   onSelect,
 }: {
-  rootLayer: Layer | undefined;
+  rootLayer?: Layer;
   selected?: string;
   onSelect?: (id?: string, options?: SelectLayerOptions) => void;
 }) {
   const [selectedLayerId, innerSelectLayer] = useState<string | undefined>();
   const [layerSelectionReason, setSelectionReason] = useState<string | undefined>();
   const [layerOverridenInfobox, setPrimitiveOverridenInfobox] = useState<OverriddenInfobox>();
   const [layers] = useState<LayerStore>(() => new LayerStore(rootLayer));
-  const [, setLayersRenderKey] = useState<number>(0);
+  const forceUpdate = useUpdate();
 
   useMemo(() => {
```

Comment:
```suggestion
  useEffect(() => {
```
Incorrect usage of useMemo